ivre/active/data.py

```diff
@@ -424,6 +425,17 @@ def merge_scanner_scripts(
         nports = len(ports["ports"])
         res["ports"][proto]["count"] = nports
         res["ports"]["count"] = res["ports"].get("count", 0) + nports
+        # If the list of scanned ports is almost exactly the same as
+        # the list of ports scanned by default by a scanner, assume it
+        # is.
+        for (dports, scanner_name) in DEFAULT_SCANNED_PORTS.get(proto, []):
+            if scanner_name not in res.get("scanners", []):
```

Comment:
Maybe you could remove this test, since you use `res.setdefault("scanners", {}).setdefault(scanner_name, set()).add(...)`

Reply:
If I read the code correctly, that's necessary to not re-add the value to the set (in the add()) if it exists.